### ISSUE TITLE
In the nic config the network is identified by name, not uuid

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#418]
Originally reported of Google Code with ID 362.

```
Since networks are indexed by uuid, they also should be identified by uuid.
Then they should be passed around as an ancillary object for operations, so that also the name and other parameters can be passed.

Proposal:
1) change nic.network to be the network uuid
2) when passing an instance to an rpc, also pass the relevant networks
3) profit
  (then the name, but also the gateway, subnet mask, etc can be read by the backend, and also for example passed to the OS scripts environment, etc).
  (also the network can be easily renamed and such)

thanks,

Guido

```
Originally added on 2013-02-07 13:05:46 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2013-02-07 13:06:09 +0000 UTC.

Added Labels: Type-Defect
Added to Milestone: Release2.7

**Comment by Ganeti-Issues-Migrator** [comment]
```
Hi Dimitris,

What do you think about this one? Could we fix at least the python side?
We can take a stab at the haskell part afterwards.

Thanks a lot,

Guido

```
Originally added on 2013-02-11 15:35:43 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Hi,

OK. I'll dig into that as soon as possible. It should be on top of devel-2.7 right?

dimara
```
Originally added on 2013-02-11 15:47:40 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Yes, I believe so, although if changes are too drastic we can even live with 2.7 using names and 2.8 using UUIDs. But it'd be better to fix it before the feature reaches stable.
Thanks,

Guido

```
Originally added on 2013-02-12 11:27:13 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Hi,

this should be considered fixed right?

dimara
```
Originally added on 2013-02-18 16:39:53 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
In principle, yes, but I found a problem with gnt-instance: https://code.google.com/p/ganeti/issues/detail?id=378
Can you have a look at that? Thx!
```
Originally added on 2013-02-19 13:03:08 +0000 UTC.

Mentioned Issue with GoogleCode ID 378 has been migrated to GitHub with ID #434.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Fixed with:

commit 9394f4d1d52710e83508884a4193aab7b47a2fe1
Author: Dimitris Aragiorgis <dimara@grnet.gr>
Date:   Tue Feb 19 18:28:42 2013 +0200

    Minor fixes regarding nic.network change
    
    Make LookupNetwork() return None in case target is None. This fixes
    Issue 380. Rapi passes network=None and the lookup should not fail.
    
    Make network client aware of new nic.network.gnt-network info shows
    the IPs of each instance inside the network.  It parses nic.networks
    field of QueryInstances() output, which now is a list of uuids an
    not names.
    
    Signed-off-by: Dimitris Aragiorgis <dimara@grnet.gr>
    Reviewed-by: Helga Velroyen <helgav@google.com>

commit c8f4e4256014033c351f3cda857068e84b6f0656
Author: Dimitris Aragiorgis <dimara@grnet.gr>
Date:   Tue Feb 19 16:45:30 2013 +0200

    Fix issue 378
    
    In case a NIC is not inside a network then netinfo None. Thus
    netinfo["name"] fails.
    
    Signed-off-by: Dimitris Aragiorgis <dimara@grnet.gr>
    Reviewed-by: Helga Velroyen <helgav@google.com>

commit 03be57a9925b2f268ab3466aafb7ccf3a5e45252
Author: Michael Hanselmann <hansmi@google.com>
Date:   Thu Feb 14 16:55:07 2013 +0100

    NEWS: Mention multi-alloc and fix typo
    
    As far as I could see, the ability to allocate multiple instances via
    RAPI was not mentioned at all. A superfluous word is also removed.
    
    Signed-off-by: Michael Hanselmann <hansmi@google.com>
    Reviewed-by: Guido Trotter <ultrotter@google.com>

commit f032d55c0dd22ecaeae4553b9cd1d85d386cb71b
Author: Dimitris Aragiorgis <dimara@grnet.gr>
Date:   Sat Feb 16 15:19:40 2013 +0200

    Add cfgupgrade for changing nic.network to uuid
    
    This gets all network info from config_data and parses
    all instances and their NICs and makes the substitution.
    
    Signed-off-by: Dimitris Aragiorgis <dimara@grnet.gr>
    Reviewed-by: Helga Velroyen <helgav@google.com>

commit aee9a8a08e25eb70086eaa7f3eb63e88e62f2ffb
Author: Dimitris Aragiorgis <dimara@grnet.gr>
Date:   Sat Feb 16 15:19:39 2013 +0200

    Remove useless code in backend for network hooks
    
    In backend NICs arrive with netinfo filled. If nic.network is not None
    nic.netinfo is not too. Thus all the info is derived from HooksDict()
    and nic.network must not be checked.
    
    Signed-off-by: Dimitris Aragiorgis <dimara@grnet.gr>
    Reviewed-by: Helga Velroyen <helgav@google.com>

commit 68c13560c36c08270e5fbcd11b33a106c27be2f6
Author: Dimitris Aragiorgis <dimara@grnet.gr>
Date:   Sat Feb 16 15:19:38 2013 +0200

    Show network name and not uuid in instance info
    
    This was before the case too. Now is bit trickier because nic.network
    is uuid. Info must derive from nic.netinfo.
    
    Signed-off-by: Dimitris Aragiorgis <dimara@grnet.gr>
    Reviewed-by: Helga Velroyen <helgav@google.com>

commit 1635eec0dc1ee572d7110f61b21a12c9b9242a1f
Author: Dimitris Aragiorgis <dimara@grnet.gr>
Date:   Sat Feb 16 15:19:37 2013 +0200

    Add IQ_NETWORKS in query tests
    
    Signed-off-by: Dimitris Aragiorgis <dimara@grnet.gr>
    Reviewed-by: Helga Velroyen <helgav@google.com>

commit 271fee187b6990efcb53d9ccefc025434110ad2f
Author: Dimitris Aragiorgis <dimara@grnet.gr>
Date:   Sat Feb 16 15:19:36 2013 +0200

    Implement network locking in Instance queries
    
    This is needed in case more info than each nic's network uuid is to
    be returned. We need to lock networks to get valid data. For now
    only the name is returned as an extra field. All other can be added
    with trivial effort.
    
    Signed-off-by: Dimitris Aragiorgis <dimara@grnet.gr>
    Reviewed-by: Helga Velroyen <helgav@google.com>

commit f98e1175cc2001da476916f3c13095f0663dea75
Author: Dimitris Aragiorgis <dimara@grnet.gr>
Date:   Sat Feb 16 15:19:35 2013 +0200

    Changes in query to support nic.network as uuid
    
    Queries now return the network uuid as well as it's name. Here we
    only use info provided be LUInstanceQueryData context.
    
    Signed-off-by: Dimitris Aragiorgis <dimara@grnet.gr>
    Reviewed-by: Helga Velroyen <helgav@google.com>

commit 434b96a252b88e13a013a2939db51c3ee80481da
Author: Dimitris Aragiorgis <dimara@grnet.gr>
Date:   Sat Feb 16 15:19:34 2013 +0200

    Modify query LUs to supoprt nic.network as uuid
    
    Make _InstanceQuery gather all network info related to instance's
    NICs and in case of NETQ_INST in _NetworkQuery get all network
    uuids directly from nic.network
    
    Signed-off-by: Dimitris Aragiorgis <dimara@grnet.gr>
    Reviewed-by: Helga Velroyen <helgav@google.com>

commit 922610c996dec8cec0bb145553257081465afcb9
Author: Dimitris Aragiorgis <dimara@grnet.gr>
Date:   Sat Feb 16 15:19:33 2013 +0200

    Add GetInstanceNetworks() config method
    
    This will be needed for Instance Queries. It walks through the
    instance's NICs and returns a list network uuids that the NICs
    are attached to.
    
    Signed-off-by: Dimitris Aragiorgis <dimara@grnet.gr>
    Reviewed-by: Helga Velroyen <helgav@google.com>

commit 5a292b360f6ef81384f02e812b3347e17bf1bee5
Author: Dimitris Aragiorgis <dimara@grnet.gr>
Date:   Sat Feb 16 15:19:32 2013 +0200

    cmdlib changes to support nic.network as uuid
    
    Refactor Instance related LUs to support nic.network as
    a uuid. This removes all the unnecessary invocations to
    LookupNetwork().
    
    Signed-off-by: Dimitris Aragiorgis <dimara@grnet.gr>
    Reviewed-by: Helga Velroyen <helgav@google.com>

commit 9ccacbc873b32169f72c63cfb825f3ffb57fff0c
Author: Dimitris Aragiorgis <dimara@grnet.gr>
Date:   Sat Feb 16 15:19:31 2013 +0200

    Make network config methods take uuid as argument
    
    This will be needed in the following patches where nic.network
    will refer to network's uuid and not name.
    
    Signed-off-by: Dimitris Aragiorgis <dimara@grnet.gr>
    Reviewed-by: Helga Velroyen <helgav@google.com>

```
Originally added on 2013-02-19 17:37:18 +0000 UTC.

Mentioned Issue with GoogleCode ID 378 has been migrated to GitHub with ID #434.
Mentioned Issue with GoogleCode ID 380 has been migrated to GitHub with ID #436.
Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
Marking 2.7 bugs fixed, as 2.7.0~rc1 was released, and should be ready to be tested in production environments.
```
Originally added on 2013-05-09 05:42:48 +0000 UTC.

Changed State: Released
